parastell.py

```diff
@@ -93,36 +95,94 @@ def cubit_export(components, export, magnets):
     cubit.cmd('imprint volume all')
     cubit.cmd('merge volume all')
 
-    # Conditionally assign magnet material group
-    if magnets is not None:
-        magnet_h5m_tag = magnets['h5m_tag']
+    if export['native_meshing']: #export using native cubit meshing capabilities v2023.11+
+
+        #extract Cubit export parameters
+        anisotropic_ratio = export['anisotropic_ratio']
+        deviation_angle = export['deviation_angle']
+
+        #conditionally populate strings
+        if anisotropic_ratio is None:
+            anisotropic_ratio = 100
+        
+        if deviation_angle is None:
+            deviation_angle = 5
+
+
+        # create materials for native cubit meshing
+        for comp in components.values():
+            cubit.cmd(f'create material "{comp["h5m_tag"]}" property_group "CUBIT-ABAQUS"')
+
+        # assign components to blocks
+        for comp in components.values():
+            cubit.cmd('set duplicate block elements off')
+            cubit.cmd("block " + str(comp['vol_id']) + " add volume " + str(comp['vol_id']))
+        
+        # assign materials to blocks
+        for comp in components.values():
+            cubit.cmd("block " + str(comp['vol_id']) + " material " + ''.join(("\'",comp['h5m_tag'],"\'")))
+            
+
+        if magnets is not None: #conditionally assign material to magnets
+            
+            magnet_h5m_tag = magnets['h5m_tag']
+            
+            # create magnet material
+            cubit.cmd(f'create material "{magnet_h5m_tag}" property_group "CUBIT-ABAQUS')
+
+            # assign magnets to block
+            block_number = min(magnets['vol_id'])
+            for vol in magnets['vol_id']:
+                cubit.cmd('set duplicate block elements off')
+                cubit.cmd("block " + str(block_number) + " add volume " + str(vol))
+            
+            # assign magnet material to block
+            cubit.cmd("block " + str(block_number) + " material " + ''.join(("\'",magnet_h5m_tag,"\'")))
+        
+        #mesh the model
+        cubit.cmd("set trimesher coarse on ratio " + str(anisotropic_ratio) + " angle " + str(deviation_angle))
+        cubit.cmd("surface all scheme trimesh")
+        cubit.cmd("mesh surface all")
+
+        #export dagmc file
+        cubit.cmd(f'export cf_dagmc "{cwd + "/dagmc.h5m"}" overwrite')
+
+    else:
+
+        # Extract Cubit export parameters
+        facet_tol = export['facet_tol']
+        len_tol = export['len_tol']
+        norm_tol = export['norm_tol']
```

Comment:
Perhaps move these closer to where they are used... or maybe do without and just use `export['facet_tol']` (etc) in the two places each are used...

Reply:
moved closer to where they are used, left the rest as is for now, I can change it to use the dict if so desired, however